### PR TITLE
clean up localsettings.example.py and import from dev_settings

### PR DESCRIPTION
**Pull request by czue**
## Summary

I floated this idea on the slack chat a little while back and it seemed to have reasonable support. This is my first pass at the "easy" decisions of things that can be removed. 

One thing I overlooked when I was originally pushing the idea to eliminate this file entirely, is that some values either only exist to be overridden locally, or are not readily documented in commcare-cloud (because they're in vault files). So `localsettings.example.py` is kind of a form of external documentation and not just for dev environments.

Everything here that was removed was either:

- already set to the same value in `dev_settings` or `settings`
- optional with a straightforward value (like a number or a string)
- optional with a complex value that was also documented in commcare-cloud (e.g. [`INTERNAL_DATA`](https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2#L498))
- apparently unused or deprecated

I did NOT remove:

- flags that are commonly modified locally (e.g. `COMPRESS_ENABLED`, `COMPRESS_OFFLINE`, `BASE_ADDRESS`, `SERVER_ENVIRONMENT`)
- anything that explicitly existed to allow overriding something locally (e.g. `LOCAL_APPS`, `LOCAL_MIDDLEWARE`)
- anything that had a complex value not available somewhere else (e.g. `MAPS_LAYERS`, which is encrypted in commcare-cloud)

There are also a few things pending resolution in [this doc](https://docs.google.com/document/d/1b0KPmUvCS5evjuc2ijnKUq8DTY-i18LYcRI6nwCFwBA/edit#).

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

dev / docs only 

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by czue** [comment]
Also this kind of depends on https://github.com/dimagi/commcare-hq/pull/29625 in that a few default values like the couch config won't work out-of-the-box without it

File: localsettings.example.py [hunk 1]
```diff
@@ -1,25 +1,10 @@
...
-        'USER': 'commcarehq',
-        'PASSWORD': 'commcarehq',
-        'HOST': 'localhost',
-        'PORT': '5432',
-        'TEST': {
-            'SERIALIZE': False,
-        },
-    }
-}
+####### Database config #######
```
[the remark this answers is not in the document]
Comment (czue) [reply localsettings.example.py:3]: (

File: localsettings.example.py [hunk 2]
```diff
@@ -230,54 +108,3 @@
                                   # to 200 to prevent AssertionError: 429 != 200
                                   # test failures in development environsments.
 CCHQ_API_THROTTLE_TIMEFRAME = 10  # seconds
```
Comment (snopoke) [inline localsettings.example.py:110]: these api settings feel like they could be in dev settings

Comment (czue) [reply localsettings.example.py]: agreed, wanted to make sure they were the right ones first: https://github.com/dimagi/commcare-hq/pull/29626/commits/d47f9f45ba524c5e7286999faedeaa35aba6d471

**Review by millerdev — APPROVED** [review]
:heart: This is great!

**Review by snopoke — APPROVED** [review]

**Pull request merged**